src/methods/call.js

```diff
@@ -331,49 +331,55 @@ export function OrdinaryCallEvaluateBody(
 
       // 2. Return the result of EvaluateBody of the parsed code that is the value of F's
       //    [[ECMAScriptCode]] internal slot passing F as the argument.
-      let code = F.$ECMAScriptCode;
-      invariant(code !== undefined);
-      let context = realm.getRunningContext();
-      let c = context.lexicalEnvironment.evaluateCompletionDeref(code, F.$Strict);
-      // We are about the leave this function and this presents a join point where all non exeptional control flows
-      // converge into a single flow using the joined effects as the new state.
-      c = Functions.incorporateSavedCompletion(realm, c);
-      let joinedEffects;
-      if (c instanceof PossiblyNormalCompletion) {
-        let e = realm.getCapturedEffects(c);
-        if (e !== undefined) {
-          // There were earlier, conditional exits from the function
-          // We join together the current effects with the effects of any earlier returns that are tracked in c.
-          realm.stopEffectCaptureAndUndoEffects(c);
-        } else {
-          e = construct_empty_effects(realm);
+      let priorSavedCompletion = realm.savedCompletion;
```

Comment:
Matter of test, but I would be prefer `callerSavedCompletion`

Reply:
Feel free to change it in one of your upcoming pull requests.